edx_proctoring/api.py

```diff
@@ -881,38 +882,31 @@ def update_attempt_status(exam_id, user_id, to_status, raise_if_not_found=True,
                 cascade_effects=False
             )
 
-    # email will be send when the exam is proctored and not practice exam
-    # and the status is verified, submitted or rejected
-    should_send_status_email = (
-        exam_attempt_obj.taking_as_proctored and
-        not exam_attempt_obj.is_sample_attempt and
-        ProctoredExamStudentAttemptStatus.needs_status_change_email(exam_attempt_obj.status)
+    # call service to get course name.
+    credit_service = get_runtime_service('credit')
+    credit_state = credit_service.get_credit_state(
+        exam_attempt_obj.user_id,
+        exam_attempt_obj.proctored_exam.course_id,
+        return_course_info=True
     )
-    if should_send_status_email:
-        # trigger credit workflow, as needed
-        credit_service = get_runtime_service('credit')
 
-        # call service to get course name.
-        credit_state = credit_service.get_credit_state(
+    default_name = _('your course')
+    if credit_state:
+        course_name = credit_state.get('course_name', default_name)
+    else:
+        course_name = default_name
+        log.info(
+            "Could not find credit_state for user id %r in the course %r.",
             exam_attempt_obj.user_id,
-            exam_attempt_obj.proctored_exam.course_id,
-            return_course_info=True
-        )
-
-        default_name = _('your course')
-        if credit_state:
-            course_name = credit_state.get('course_name', default_name)
-        else:
-            course_name = default_name
-            log.info(
-                "Could not find credit_state for user id %r in the course %r.",
-                exam_attempt_obj.user_id,
-                exam_attempt_obj.proctored_exam.course_id
-            )
-        send_proctoring_attempt_status_email(
-            exam_attempt_obj,
-            course_name
+            exam_attempt_obj.proctored_exam.course_id
         )
+    email = create_proctoring_attempt_status_email(
+        user_id,
+        exam_attempt_obj,
+        course_name
+    )
+    if email:
+        email.send()
```

Comment:
That was my original intent, but then it turned out not to be necessary. I think I prefer it this way anyway, if that's okay.